workflow/controller/taskresult.go

```diff
@@ -53,21 +62,100 @@ func (wfc *WorkflowController) newWorkflowTaskResultInformer() cache.SharedIndex
 	return informer
 }
 
-func (woc *wfOperationCtx) taskResultReconciliation() {
+func shouldErrorPodTimeout(node *wfv1.NodeStatus) bool {
+	return time.Since(node.StartedAt.Time) <= envutil.LookupEnvDurationOr("POD_ABSENT_TIMEOUT", 2*time.Minute)
+}
+
+func (woc *wfOperationCtx) taskResultReconciliation() error {
+
 	objs, _ := woc.controller.taskResultInformer.GetIndexer().ByIndex(indexes.WorkflowIndex, woc.wf.Namespace+"/"+woc.wf.Name)
 	woc.log.WithField("numObjs", len(objs)).Info("Task-result reconciliation")
+
+	taskresultCli := woc.controller.wfclientset.ArgoprojV1alpha1().WorkflowTaskResults(woc.wf.Namespace)
+	// We will be hitting the cache here, should we bypass the cache?
```

Comment:
Hitting the cache feels fine, we shouldn't bypass it or this would be horrific for performance.

Suggest tidying the comment.

Reply:
Sure, can do